qcodes/instrument/parameter.py

```diff
@@ -1745,6 +1695,97 @@ def setpoint_full_names(self) -> Optional[Sequence[Sequence[str]]]:
             return self.setpoint_names
 
 
+class _Cache:
+    def __init__(self,
+                 parameter: '_BaseParameter',
+                 max_val_age: Optional[float] = None):
+        self._parameter = parameter
+        self._value: ParamDataType = None
+        self._raw_value: ParamRawDataType = None
```

Comment:
as we found out offline, it actually should. This will make `_BaseParameter` use only PUBLIC things of the `_Cache`. So, a read-only `raw_value` property has been added now to `_Cache`